contracts/src/OffChainMessageContracts/ValidatorSetSig.sol

```diff
@@ -0,0 +1,113 @@
+// (c) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// SPDX-License-Identifier: Ecosystem
+
+pragma solidity 0.8.18;
+
+import {
+    WarpMessage,
+    IWarpMessenger
+} from "@avalabs/subnet-evm-contracts@1.2.0/contracts/interfaces/IWarpMessenger.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts@4.8.1/security/ReentrancyGuard.sol";
+
+struct ValidatorSetSigMessage {
```

Comment:
As I am writing e2e tests I've realized that `abi-gen` doesn't generate go structs for this unless it is used as an argument to an external or public function.

I can either remove this convenience wrapper and decode directly as a tuple or provide a new `validateMessage` `public view` function that checks relevant arguments including nonce. The benefit of this would be that the people preparing messages can confirm that they pass validation before passing them onto validators to get signed and sent over as a `WarpMessage`.

Reply:
I think you just need to add `ValidatorSetSig` to the list in `abi_gen.sh` https://github.com/ava-labs/teleporter/blob/main/scripts/abi_bindings.sh#L19

Reply:
Is there a strong reason why this type needs to be in the ABI generated files at all? It would be helpful for constructing the Warp message in e2e tests, but I'm not convinced that that's reason enough to add the type to the ABI. 

Adding a `validateMessage` method would work as you described, but I don't know how useful it would be in the contract itself, since it would be likely used out-of-band of the actual message provided to `executeMessage`. In which case, manually verifying by other means (i.e. inspection of the `require` statements in `executeMessage) would yield the same result.

Reply:
I don't feel strongly about it. For e2e testing I can either manually define the struct in my packing.go file or I can modify the contract to not use the wrapping struct and just encode/decode arguments directly. I only included it in the first place because i have 5 arguments and wanted their uses and naming to be clear up front. 

The `validateMessage` method was a post-hoc justification and I don't feel strongly about it either. I agree that visually inspecting the code and `require`  statements would do the same thing. The utility behind this would be a "fool-proof" way to confirm that the message is valid before handing it off to validators for signing.  A practical downside here is that having another public function call where we pass the struct to the `validateMessage` is likely to consume more gas for copying the struct. I don't see this as a big problem given rarity of invocation of this contract.

All that being said my current change does include it but let me know if we want to remove it and I will go back and update the e2e tests. The options as I see them are:

:one: Leave as is with `validateMessage` method available and used by `executeCall`
:two: Remove the `validateMessage` but keep using the `ValidatorSetSigMessage` struct in the contract and manually define the matching go struct in`packing.go`
:three: Remove the `ValidatorSetSigMessage` struct from the contract and encode/decode arguments directly.

Reply:
I am leaning towards option 1, mostly from a readability and extensibility stand point. It'd make submitting transactions to this contract more clear for what parameters need to be passed in, and also easier to extend to this struct if needed. This would also fit in nicely with our current pattern for writing unit/e2e tests. 

The downside I'd see for this option would be deployment gas costs, but shouldn't too big of an impact, especially considering this contract can be used against different target addresses.